rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1759,8 +1765,17 @@ void main_window::CreateConnects()
 
 	connect(ui->toolskernel_explorerAct, &QAction::triggered, [this]
 	{
-		kernel_explorer* kernelExplorer = new kernel_explorer(this);
-		kernelExplorer->show();
+		if (!m_kernel_explorer)
+		{
+			m_kernel_explorer = new kernel_explorer(this);
+
+			connect(m_kernel_explorer, &kernel_explorer::FrameClosed, [this]()
+			{
+				m_kernel_explorer = nullptr;
```

Comment:
Doesnt this happen automagically if you close the Dialog?

Reply:
I've just now tested it whether qt does some magic in background using:
1. Execute game.
2. Execute kernel explorer.
3. Close the game. (kernel explorer close() called)
4. Reexecute the game. 
5. Rexecute kernel explorer. 

Step 5 caused a segfault because m_kernel_explorer was not a nullptr yet a dangling pointer address. So I assume it does not happen.

Reply:
I've just now tested it without this code whether qt does some magic in background using:
1. Execute game.
2. Execute kernel explorer.
3. Close the game. (kernel explorer close() called)
4. Reexecute the game. 
5. Rexecute kernel explorer. 

Step 5 caused a segfault because m_kernel_explorer was not a nullptr yet a dangling pointer address. So I assume it does not happen.

Reply:
Also raw pointer in c++ is just a value, it is not attached to any object unlike std::unique_ptr so I can't see how qt can detect its attachement to any object.